### PR TITLE
Advanced Energy Gun revamp

### PR DESCRIPTION
**Pull request by DeityLink**
Sprites by @otatoh 

https://cdn.discordapp.com/attachments/246096207523348481/603788270303707136/dreamseeker_2019-07-25_05-12-53-62.webm
image
image

:cl:
* rscadd: The Advanced Energy Gun gets a brand new look courtesy of otatoh. The old sprite still exists and an "experimental" version using it can be spawned by admins.
* bugfix: Fixed the old AEG's in-hand sprite never properly updating as its charge diminishes.
* tweak: Changed the way AEGs react to EMPs, they now have 10 points of "core stability". When getting EMP'd, they lose 4 points if the EMP is strong, and 2 if it's weak. This affects their recharge rate (which is 10 times the core stability, so 100 charge by default every 4 seconds.), and creates a radiation burst once it reaches zero. The radiation now always reaches up to 4 tiles away, but the amount received decreases with the distance (300 rad on the same tile and adjacent, then divided by the distance from the gun). Previously a critical AEG would keep producing deadly radiation burst every few seconds. If the AEG lost some stability but it still is over zero, it has a percentage chance equal to its current core stability to recover some of it every second. AEGs no longer have a chance to produce radiation bursts as long as they have some stability left, otherwise the gun becomes unusable.

### CODE REVIEW

**Comment by DeityLink** [comment]
Also making ERT spawn with those instead of the regular energy gun because seriously why, they have half a dozen of those waiting at their base.

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 1]
```diff
@@ -42,98 +45,131 @@
 				playsound(user,'sound/weapons/egun_toggle_noammo.ogg',73,0,-5)
 	update_icon()
 
+////////////////////////////ADVANCED ENERGY GUN////////////////////////////////////////////////////
+
 /obj/item/weapon/gun/energy/gun/nuclear
-	name = "Advanced Energy Gun"
-	desc = "An energy gun with an experimental miniaturized reactor."
+	name = "\improper Advanced Energy Gun"
```
Comment (Kurfursten) [inline code/modules/projectiles/guns/energy/nuclear.dm:51]: no real reason this shouldn't just be converted to lowercase entirely

Comment (DeityLink) [reply code/modules/projectiles/guns/energy/nuclear.dm:51]: I kinda like the capitalization, and it's how it's always been. This way at least now we get "the".

**Comment by SonixApache** [comment]
Huh.
I wanted to do this, but instead of the giving science 'production-grade' sprite, I was going to give it exclusively to the ERT with a reduced capacity and recharge rate.
Pretty sure Blithering made a similar sprite at some point.

Only reason I didn't do it is because I wanted to replace the stupid gun racks with Unreal/Quake style weapon spawners.

**Comment by DeityLink** [comment]
>I was going to give it exclusively to the ERT with a reduced capacity and recharge rate.

You realize AEGs already have a pretty shit recharge rate right?

**Comment by Blithering** [comment]
I made a similar sprite, but otatoh's one is much better. Any chance you can add USPs to the ERT gear warehouse while you're poking around there? https://github.com/vgstation-coders/vgstation13/issues/16609

**Comment by DeityLink** [comment]
@Blithering on one hand that would be even less atomic, on the other hand it would also require map changes (which I'm not doing either in this PR). So two reasons to decline here.

**Comment by Blithering** [comment]
Fair enough, I forgot the ERT regular eguns are equipped on spawn loadout rather than mapped in.

**Comment by DrSnips** [comment]
Could you update the OP with some of the EMP and stability numbers please? "When it hits 0" isn't a lot to go off of for everyone. The changes are fine but it should be a little bit more transparent.

**Comment by DeityLink** [comment]
@DrSnips added info to OP.

TL;DR: it takes at least 3 EMPs to break an AEG. Up to 5 if the EMPs were weak.

**Comment by Chazach** [comment]
sprite nice
the one counter to people with 5 trillion AEGs being nerfed bad

**Comment by DeityLink** [comment]
@Chazach Why? they still get a deadly dose of radiation if they're hit by 3 ion rifle shots.

**Comment by FudgePucker** [comment]
Why would science have better recharging eguns over ERT anyways?

**Comment by SonixApache** [comment]
Because it's cutting-edge experimental science, unlike the mass-produced slightly-higher-quality-than-usual NT guns

**Comment by DeityLink** [comment]
ERT also had those, I don't get the point of the discussion.
